parlai/core/agents.py

```diff
@@ -154,6 +156,78 @@ def shutdown(self):
         """
         pass
 
+    def self_observe(self, self_message: Message) -> None:
```

Comment:
Let's not add this for now and use a hasattr to check if self-observe should be called

Reply:
Okay sure!